core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java

```diff
@@ -394,4 +413,54 @@ private CloudSqlInstance apiFetcher(ConnectionConfig config) {
         localKeyPair,
         RateLimiter.create(1.0 / 30.0)); // 1 refresh attempt every 30 seconds
   }
+
+  ConnectionConfig resolveConfig(ConnectionConfig config) {
+    if (config.getNamedConnection() != null) {
+      ConnectionConfig namedConfig = namedConfigs.get(config.getNamedConnection());
+      if (namedConfig == null) {
+        throw new IllegalArgumentException(
+            "No named connection named " + config.getNamedConnection());
+      }
+      return namedConfig;
+    }
+    return config;
+  }
+
+  @Override
+  public void register(String name, ConnectionConfig config) {
+    ConnectionConfig oldConfig = this.namedConfigs.putIfAbsent(name, config);
+    if (oldConfig != null) {
+      CloudSqlInstance oldInstance = this.namedInstances.get(name);
+      if (oldInstance != null) {
+        oldInstance.close();
+      }
+    }
+  }
+
+  @Override
+  public void close(String name) {
```

Comment:
Same thing here: closing an unknown connector should throw an exception.

Reply:
Fixed.